src/main/java/com/elixr/poc/service/PurchaseService.java

```diff
@@ -20,17 +22,25 @@ public PurchaseService(PurchaseRepository purchaseRepository) {
      *
      * @param purchaseId
      * @return
-     * @throws NoRecordFoundException
+     * @throws GlobalException
      */
-    public boolean deletePurchaseDetails(UUID purchaseId) throws NoRecordFoundException {
+    public boolean deletePurchaseDetails(UUID purchaseId) throws GlobalException {
         boolean success = false;
         boolean purchaseRecordExists = purchaseRepository.existsById(purchaseId);
         if (purchaseRecordExists) {
             purchaseRepository.deleteById(purchaseId);
             success = true;
         } else {
-            throw new NoRecordFoundException(ApplicationConstants.ID_MISMATCH);
+            throw new GlobalException(ApplicationConstants.ID_MISMATCH);
         }
         return success;
     }
+
+    /**
+     * Finding Purchase by purchaseId and returning the purchase.
+     */
+    public Purchase getPurchaseByPurchaseId(UUID purchaseId) throws GlobalException {
+        Optional<Purchase> purchase = purchaseRepository.findById(purchaseId);
+        return purchase.orElseThrow(() -> new GlobalException(ApplicationConstants.ID_MISMATCH));
```

Comment:
The error message should match the use case.

Reply:
Updated